include/sundials/sundials_control.h

```diff
@@ -0,0 +1,248 @@
+ *    MRI_H   - controls two multirate step sizes
+ *    MRI_TOL - controls slow and fast relative tolerances
+ * ----------------------------------------------------------------- */
+
+typedef enum
+{
+  SUNDIALS_CONTROL_NONE,
+  SUNDIALS_CONTROL_H,
+  SUNDIALS_CONTROL_HQ,
+  SUNDIALS_CONTROL_MRI_H,
+  SUNDIALS_CONTROL_MRI_TOL
+} SUNControl_ID;
+
+/* -----------------------------------------------------------------
+ * Generic definition of SUNControl
+ * ----------------------------------------------------------------- */
+
+/* Forward reference for pointer to SUNControl_Ops object */
+typedef _SUNDIALS_STRUCT_ _generic_SUNControl_Ops* SUNControl_Ops;
+
+/* Forward reference for pointer to SUNControl object */
+typedef _SUNDIALS_STRUCT_ _generic_SUNControl* SUNControl;
+
+/* Structure containing function pointers to controller operations  */
+struct _generic_SUNControl_Ops
+{
+  /* REQUIRED of all controller implementations. */
+  SUNControl_ID (*getid)(SUNControl C);
+  void (*destroy)(SUNControl C);
+
+  /* REQUIRED for controllers of SUNDIALS_CONTROL_H type. */
+  int (*estimatestep)(SUNControl C, realtype h,
+                      realtype dsm, realtype* hnew);
+
+  /* REQUIRED for controllers of SUNDIALS_CONTROL_HQ type. */
+  int (*estimatestepandorder)(SUNControl C, realtype h, int q,
+                              realtype dsm, realtype* hnew, int *qnew);
+
+  /* REQUIRED for controllers of SUNDIALS_CONTROL_MRI_H type. */
+  int (*estimatemristeps)(SUNControl C, realtype H, realtype h,
+                          realtype DSM, realtype dsm,
+                          realtype* Hnew, realtype *hnew);
+
+  /* REQUIRED for controllers of SUNDIALS_CONTROL_MRI_TOL type. */
+  int (*estimatesteptol)(SUNControl C, realtype H, realtype tolfac,
+                         realtype DSM, realtype dsm, realtype *Hnew,
+                         realtype* tolfacnew);
+
+  /* OPTIONAL for all SUNControl implementations. */
+  int (*reset)(SUNControl C);
+  int (*setdefaults)(SUNControl C);
+  int (*write)(SUNControl C, FILE* fptr);
+  int (*setmethodorder)(SUNControl C, int q);
+  int (*setembeddingorder)(SUNControl C, int p);
+  int (*seterrorbias)(SUNControl C, realtype bias);
+  int (*update)(SUNControl C, realtype h, realtype dsm);
+  int (*updatemrih)(SUNControl C, realtype H, realtype h,
+                    realtype DSM, realtype dsm);
+  int (*updatemritol)(SUNControl C, realtype H, realtype tolfac,
+                      realtype DSM, realtype dsm);
+  int (*space)(SUNControl C, long int *lenrw, long int *leniw);
+#ifdef __cplusplus
+  _generic_SUNControl_Ops() = default;
+#endif
+
+};
+
+/* A SUNControl is a structure with an implementation-dependent
+   'content' field, and a pointer to a structure of
+   operations corresponding to that implementation. */
+struct _generic_SUNControl
+{
+  void* content;
+  SUNControl_Ops ops;
+  SUNContext sunctx;
+#ifdef __cplusplus
+  _generic_SUNControl() = default;
+#endif
```

Comment:
I think this can be removed (see #329)

Reply:
These were all inserted to emulate what is done in `SUNMemory`.  I'll remove them from this PR, but perhaps the others should be removed as well.